apps/namada-interface/src/services/extensionEvents/handlers/namada.ts

```diff
@@ -45,6 +45,9 @@ export const NamadaTxStartedHandler =
 export const NamadaTxCompletedHandler =
   (dispatch: Dispatch<unknown>) => async (event: CustomEventInit) => {
     const { msgId, txType, success, payload } = event.detail;
+    if (payload) {
```

Comment:
It might be more future-proof to check `success` instead of `payload` for a failed transaction in case we want to send a payload for a successful transaction at some point.

Reply:
fixed!